drivers/display/display_max7219.c

```diff
@@ -0,0 +1,385 @@
+	int ret;
+
+	if (!spi_is_ready(&dev_config->spi)) {
+		LOG_ERR("SPI device not ready");
+		return -ENODEV;
+	}
+
+	/* turn off all leds */
+	memset(dev_data->digit_buf, 0,
+	       dev_config->num_cascading * MAX7219_DIGITS_PER_DEVICE * sizeof(uint8_t));
+
+	ret = max7219_transmit_all(dev, MAX7219_REG_DISPLAY_TEST, MAX7219_LEAVE_DISPLAY_TEST_MODE);
+	if (ret < 0) {
+		LOG_ERR("Failed to disable display test");
+		return ret;
+	}
+
+	ret = max7219_transmit_all(dev, MAX7219_REG_DECODE_MODE, MAX7219_NO_DECODE);
+	if (ret < 0) {
+		LOG_ERR("Failed to set decode mode");
+		return ret;
+	}
+
+	ret = max7219_transmit_all(dev, MAX7219_REG_INTENSITY, dev_config->intensity);
+	if (ret < 0) {
+		LOG_ERR("Failed to set global brightness");
+		return ret;
+	}
+
+	ret = max7219_transmit_all(dev, MAX7219_REG_SCAN_LIMIT, dev_config->scan_limit);
+	if (ret < 0) {
+		LOG_ERR("Failed to set scan limit");
+		return ret;
+	}
+
+	ret = max7219_transmit_all(dev, MAX7219_REG_SHUTDOWN, MAX7219_LEAVE_SHUTDOWN_MODE);
+	if (ret < 0) {
+		LOG_ERR("Failed to leave shutdown state");
+		return ret;
+	}
+
+	const struct display_buffer_descriptor desc = {
+		.buf_size = dev_config->num_cascading * MAX7219_DIGITS_PER_DEVICE,
+		.height = dev_config->num_cascading * MAX7219_DIGITS_PER_DEVICE,
+		.width = MAX7219_DIGITS_PER_DEVICE,
+		.pitch = MAX7219_DIGITS_PER_DEVICE,
+	};
+
+	ret = max7219_write(dev, 0, 0, &desc, dev_data->digit_buf);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+#define DISPLAY_MAX7219_INIT(n)                                                                    \
+	static uint8_t max7219_digit_data_##n[DT_INST_PROP(n, num_cascading) *                     \
+					      MAX7219_DIGITS_PER_DEVICE];                          \
+	static uint8_t max7219_tx_buf##n[DT_INST_PROP(n, num_cascading) * 2];                      \
+	static struct max7219_data max7219_data_##n = {                                            \
+		.digit_buf = max7219_digit_data_##n,                                               \
+		.tx_buf = max7219_tx_buf##n,                                                       \
+	};                                                                                         \
+	static const struct max7219_config max7219_config_##n = {                                  \
+		.spi = SPI_DT_SPEC_INST_GET(n, SPI_OP_MODE_MASTER | SPI_WORD_SET(8U), 0U),         \
+		.num_cascading = DT_INST_PROP(n, num_cascading),                                   \
+		.intensity = DT_INST_PROP(n, intensity),                                           \
+		.scan_limit = DT_INST_PROP(n, scan_limit),                                         \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(n, max7219_init, NULL, &max7219_data_##n, &max7219_config_##n,       \
+			      POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY, &max7219_api);
+
+DT_INST_FOREACH_STATUS_OKAY(DISPLAY_MAX7219_INIT)
```

Comment:
please indent consistently with tabs

```diff
diff --git a/drivers/display/display_max7219.c b/drivers/display/display_max7219.c
index e79c1cb6bee..00e4dae1c55 100644
--- a/drivers/display/display_max7219.c
+++ b/drivers/display/display_max7219.c
@@ -365,21 +365,25 @@ static int max7219_init(const struct device *dev)
        return 0;
 }
 
-#define DISPLAY_MAX7219_INIT(n)                                                                    \
-       static uint8_t max7219_digit_data_##n[DT_INST_PROP(n, num_cascading) *                     \
-                                             MAX7219_DIGITS_PER_DEVICE];                          \
-       static uint8_t max7219_tx_buf##n[DT_INST_PROP(n, num_cascading) * 2];                      \
-       static struct max7219_data max7219_data_##n = {                                            \
-               .digit_buf = max7219_digit_data_##n,                                               \
-               .tx_buf = max7219_tx_buf##n,                                                       \
-       };                                                                                         \
-       static const struct max7219_config max7219_config_##n = {                                  \
-               .spi = SPI_DT_SPEC_INST_GET(n, SPI_OP_MODE_MASTER | SPI_WORD_SET(8U), 0U),         \
-               .num_cascading = DT_INST_PROP(n, num_cascading),                                   \
-               .intensity = DT_INST_PROP(n, intensity),                                           \
-               .scan_limit = DT_INST_PROP(n, scan_limit),                                         \
-       };                                                                                         \
-       DEVICE_DT_INST_DEFINE(n, max7219_init, NULL, &max7219_data_##n, &max7219_config_##n,       \
-                             POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY, &max7219_api);
+#define DISPLAY_MAX7219_INIT(n)                                                        \
+       static uint8_t max7219_digit_data_##n[DT_INST_PROP(n, num_cascading) *  \
+                                             MAX7219_DIGITS_PER_DEVICE];       \
+       static uint8_t max7219_tx_buf##n[DT_INST_PROP(n, num_cascading) * 2];   \
+       static struct max7219_data max7219_data_##n = {                         \
+               .digit_buf = max7219_digit_data_##n,                            \
+               .tx_buf = max7219_tx_buf##n,                                    \
+       };                                                                      \
+       static const struct max7219_config max7219_config_##n = {               \
+               .spi = SPI_DT_SPEC_INST_GET(n, SPI_OP_MODE_MASTER |             \
+                                           SPI_WORD_SET(8U), 0U),              \
+               .num_cascading = DT_INST_PROP(n, num_cascading),                \
+               .intensity = DT_INST_PROP(n, intensity),                        \
+               .scan_limit = DT_INST_PROP(n, scan_limit),                      \
+       };                                                                      \
+                                                                               \
+       DEVICE_DT_INST_DEFINE(n, max7219_init, NULL,                            \
+                             &max7219_data_##n, &max7219_config_##n,           \
+                             POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY,        \
+                             &max7219_api);
 
 DT_INST_FOREACH_STATUS_OKAY(DISPLAY_MAX7219_INIT)
```

Reply:
Thanks. I have used the clang-format format this file. Should I add clang-format off for this block? or follow clang-format config in zephyr?

Reply:
> Thanks. I have used the `clang-format` format this file. Do you also use the `clang-format`?

No, it is just Vim and me.